model.py

```diff
@@ -4178,6 +4178,21 @@ class LicensePoolDeliveryMechanism(Base):
     # One LicensePoolDeliveryMechanism may fulfill many Loans.
     fulfills = relationship("Loan", backref="fulfillment")
 
+    # One LicensePoolDeliveryMechanism may be associated with one RightsStatus.
+    rightsstatus_id = Column(
+        Integer, ForeignKey('rightsstatus.id'), index=True)
+
+
+    def set_rights_status(self, uri, name=None):
+        _db = Session.object_session(self)
+        status, ignore = get_one_or_create(
+            _db, RightsStatus, uri=uri,
+            create_method_kwargs=dict(name=name))
+        self.rights_status = status
+        if status.uri in RightsStatus.OPEN_ACCESS:
+            self.license_pool.open_access = True
```

Comment:
Presumably we should also set open_access to False if we just destroyed the last remaining open-access delivery mechanism, but I'm not too worried about that case.